sympy/stats/crv.py

```diff
@@ -173,8 +336,21 @@ def check(*args):
 
     def sample(self, size=()):
         """ A random realization from the distribution """
-        icdf = self._inverse_cdf_expression()
         if not size:
+            size = 1
+        samps = getattr(SampleExternal, '_sample_python')(self, size)
+        if samps is not None:
+            return samps if size!=1 else samps[0]
+
+        libraries = ['scipy', 'numpy', 'pymc3']
```

Comment:
I think that it can get ambiguous which library it is using for sampling.
I see it already return different types of arrays.

So I think that it should better be specified as an option and unsupport stuff that are not available for specific libraries, than trying all the libraries to get it succeed.

Reply:
In my opinion, at the user end, we don't need to specify which library is used to sample. Also, for coverage purposes, we can do by directly calling the concerned method from class `SampleExternal`. IMO, We should not go for unsupported kind of stuff, as providing such an option, also will lead to unnecessary raising errors, instead, I suggest just return the sample without any specification of the library. To maintain the uniformity, I will type-cast each of the samples(in case more than one sample is required) returned by different libraries to `list`. Please share your thoughts @czgdp1807.

Reply:
I think, there should be an option provided to the user for using libraries. In this way user will be knowing the type of the output they will be getting. We should keep the public API unambiguous. Type casting is not recommended when we can give an option and make the things easy.

Reply:
Okay. I will go with the suggestion. As there is a change in the public API of the stats module, I will initially work for the continuous RV.
* Firstly, it will require addition of an argument in each function call of `sample`.
* I will be keeping this as`None` by default for all the calls not used in Continuous RV.
After this gets finalized for Continuous RV, I will open a new PR for the same changes in Finite and Discrete RV, and will make the actual use of lib argument in their function calls also, so that the diff size remains under some bound and becomes easy to be reviewed in parts.
Does this plan look fine?

Reply:
What used to happen earlier before this framework was introduced?
May be to avoid change in the public API, library can be passed as keyword argument and if nothing is given as library then inbuilt python sampling features can be used.

Reply:
Yes, I meant the same. There was no library argument before, so, by default will be python. But, I wanted to ask that firstly, should I implement this argument usage for Continuous RV, and then make this use of argument for Finite and Discrete?

Reply:
This PR is concerned with CRV, so first we should try out the approach with CRVs and then we can proceed with  FRV and DRV.

Reply:
Thanks! I will make the necessary changes.